src/tools/sdk/dart_sdk_archive/lib/src/version_selector.dart

```diff
@@ -144,13 +144,30 @@ class VersionSelector {
           }
         }
 
+        // No Mac 32-bit SDK builds after 2.80
         if (name == 'macOS' && platformVariant.architecture == 'ia32') {
-          // No Mac 32-bit SDK builds >= 2.80
           if (versionInfo.version > Version(2, 7, 0)) {
             continue;
           }
         }
 
+        // No Mac arm64 SDK builds before 2.14.0, after July 1st 2021, and not in stable yet.
+        // TODO: After 2.14 simply change these tests to versionInfo.version >= Version(2,14,0).
```

Comment:
What about simply checking for `>= 2.14.0-216.0.dev` (the first dev version that had signed builds)?

We could also pick any arbitrary later dev version (`2.14.0-281.0.dev` would be the first in July that was published to dart.dev).

Then it could be:

```dart
versionInfo.version >= Version(2, 14, 0, pre: '281.0.dev')
```

Older beta versions would be before that and all 2.13.x stable versions as well so we wouldn't need the date or stable channel check.

Reply:
PR updated, PTAL